src/User32/User32.cs

```diff
@@ -2921,6 +2921,381 @@ public static partial class User32
+        /// <param name="bMenu">Indicates whether the window has a menu.</param>
+        /// <param name="dwExStyle">The Extended Window Style of the window whose required size is to be calculated.</param>
+        /// <param name="dpi">The DPI to use for scaling.</param>
+        /// <returns>
+        /// If the function succeeds, the return value is true.
+        /// If the function fails, the return value is false. To get extended error information, call <see cref="GetLastError"/>.
+        /// </returns>
+        [DllImport(nameof(User32), SetLastError = true, CharSet = CharSet.Unicode)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern unsafe bool AdjustWindowRectExForDpi(
+            [In][Out] RECT* lpRect,
+            [In] WindowStyles dwStyle,
+            [In] [MarshalAs(UnmanagedType.Bool)] bool bMenu,
+            [In] WindowStylesEx dwExStyle,
+            [In] int dpi);
+
+        /// <summary>
+        /// Determines whether two <see cref="DPI_AWARENESS_CONTEXT"/> values are identical.
+        /// </summary>
+        /// <param name="dpiContextA">The first value to compare.</param>
+        /// <param name="dpiContextB">The second value to compare.</param>
+        /// <returns>Returns true if the values are equal, otherwise false.</returns>
+        /// <remarks>
+        /// A <see cref="DPI_AWARENESS_CONTEXT"/> contains multiple pieces of information. For example, it includes both the current and the inherited <see cref="DPI_AWARENESS"/> values. AreDpiAwarenessContextsEqual ignores informational flags and determines if the values are equal. You can't use a direct bitwise comparison because of these informational flags.
+        /// </remarks>
+        [DllImport(nameof(User32), CharSet = CharSet.Unicode)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool AreDpiAwarenessContextsEqual(
+            [In] DPI_AWARENESS_CONTEXT dpiContextA,
+            [In] DPI_AWARENESS_CONTEXT dpiContextB);
+
+        /// <summary>
+        /// In high-DPI displays, enables automatic display scaling of the non-client area portions of the specified top-level window. Must be called during the initialization of that window.
+        /// </summary>
+        /// <param name="hwnd">The window that should have automatic scaling enabled.</param>
+        /// <returns>
+        /// If the function succeeds, the return value is true.
+        /// If the function fails, the return value is false. To get extended error information, call <see cref="GetLastError"/>.
+        /// </returns>
+        [DllImport(nameof(User32), SetLastError = true, CharSet = CharSet.Unicode)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern unsafe bool EnableNonClientDpiScaling([In] void* hwnd);
```

Comment:
We type our HWND handles as `IntPtr` everywhere because they are *never* dereferenced in user code and thus needn't be expressed as *pointers* anywhere. If you change this to match, HWND handles will be easier to pass around this library across this API to others.
Here and anywhere else you pass HWND around.